src/System.Text.Primitives/System/Text/EncodingData.cs

```diff
@@ -365,8 +346,196 @@ private bool VerifySuffix(ReadOnlySpan<byte> buffer, int codeUnitIndex, Symbol s
         // CC = the max value, and DD = the max value's index in the same coord-system as BB.
         struct ParsingTrieNode
         {
-            public byte valueOrNumChildren;
+            public byte ValueOrNumChildren;
             public int IndexOrSymbol;
         }
+
+        private class AbstractParsingTrie
+        {
+            private interface Node
```

Comment:
It would be nice to add a comment that this interface is added just for code clarity, but otherwise useless.